torch/csrc/jit/tensorexpr/cuda_half_support.h

```diff
@@ -85,12 +85,32 @@ class CudaHalfScalarRewriter : public IRMutator {
   const Expr* mutate(const Var* v) override {
     auto it = var_map.find(v);
     if (it != var_map.end()) {
-      return it->second;
+      return new Cast(kHalf, it->second);
     }
 
     return v;
   }
 
+  const Expr* mutate(const Cast* v) override {
```

Comment:
do we typically keep this kind of peepholes in individual passes or move them to simplifier?

Reply:
Normally they'd be in the simplifier, but I'm not confident we always want to eliminate Float(Half(X)) since it does have an affect (reducing precision). What do you think.